### ISSUE TITLE
Add support for BIT_AND_AGG(), BIT_OR_AGG(), BIT_XOR_AGG()

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10036]
Oracle 20c introduces new aggregate functions:

- `BIT_AND_AGG()`
- `BIT_OR_AGG()`
- `BIT_XOR_AGG()`

This could be emulated by producing a big expression for each bit in a data type:

```sql
-- Native support, assuming a is a TINYINT
SELECT bit_and_agg(a) FROM t;

-- Ignoring NULL for simplicity of the example
SELECT
    CASE WHEN every(bit_and(a,   1) != 0) THEN   1 ELSE 0 END
  + CASE WHEN every(bit_and(a,   2) != 0) THEN   2 ELSE 0 END
  + CASE WHEN every(bit_and(a,   4) != 0) THEN   4 ELSE 0 END
  + CASE WHEN every(bit_and(a,   8) != 0) THEN   8 ELSE 0 END
  + CASE WHEN every(bit_and(a,  16) != 0) THEN  16 ELSE 0 END
  + CASE WHEN every(bit_and(a,  32) != 0) THEN  32 ELSE 0 END
  + CASE WHEN every(bit_and(a,  64) != 0) THEN  64 ELSE 0 END
  + CASE WHEN every(bit_and(a, 128) != 0) THEN 128 ELSE 0 END
FROM t;
```

The expression would be even more complex if `EVERY()` is being emulated for a dialect.

The `BIT_OR_AGG()` emulation is similar to `BIT_AND_AGG()`. `BIT_XOR_AGG()` needs to count all bits and check if sum is even or odd.

Native implementations are available in:

- [x] BigQuery: https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#bit_and, etc.
- [x] CockroachDB: https://www.cockroachlabs.com/docs/v21.1/functions-and-operators.html
- [x] H2: http://h2database.com/html/functions-aggregate.html#bit_and
- [x] Ignite: https://ignite.apache.org/docs/latest/sql-reference/aggregate-functions
- [x] MariaDB: https://mariadb.com/kb/en/bit_and/
- [x] MySQL: https://dev.mysql.com/doc/refman/8.0/en/aggregate-functions.html#function_bit-and
- [x] Oracle: https://docs.oracle.com/en/database/oracle/oracle-database/20/newft/bitwise-aggregate-functions.html
- [x] PostgreSQL 14: https://www.postgresql.org/docs/14/functions-aggregate.html
- [x] Redshift: https://docs.aws.amazon.com/redshift/latest/dg/c_bitwise_aggregate_functions.html
- [x] Snowflake: https://docs.snowflake.com/en/sql-reference/expressions-byte-bit.html
- [x] Sybase SQL Anywhere: http://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.help.sqlanywhere.12.0.1/dbreference/bit-and-function.html
- [x] Vertica: https://www.vertica.com/docs/9.2.x/HTML/Content/Authoring/SQLReferenceManual/Functions/Aggregate/BIT_AND.htm?tocpath=SQL%20Reference%20Manual%7CSQL%20Functions%7CAggregate%20Functions%7C_____8

Though these dialects don't support `BIT_XOR_AGG`:

- [x] CockroachDB
- [x] H2
- [x] Ignite
- [x] Redshift

We'll support this:

- [x] In the API (both aggregate and window functions)
- [x] As emulations
- [x] In the parser (parse dialect disambiguation is required for `BIT_AND()`, `BIT_OR()`, and `BIT_XOR()`)

----
See also: https://docs.oracle.com/en/database/oracle/oracle-database/20/newft/bitwise-aggregate-functions.html

**Comment by lukaseder** [comment]
BigQuery also has a native implementation for these.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.15. The PostgreSQL 14 improvement will be done when PG 14 is available: https://github.com/jOOQ/jOOQ/issues/11957
